src/classes/RD2_RecurringDonations_TDTM.cls

```diff
@@ -382,14 +382,42 @@ public class RD2_RecurringDonations_TDTM extends TDTM_Runnable {
         RD2_QueueableService.EvaluateInstallmentOpportunities service =
             new RD2_QueueableService.EvaluateInstallmentOpportunities(rdIds, isScheduleChangedById);
 
-        if (Limits.getQueueableJobs() < Limits.getLimitQueueableJobs()) {
-            System.enqueueJob(service);
+        if (triggerAction == TDTM_Runnable.Action.AfterInsert && evaluateOppsSynchronously()) {
+            service.executeOppEvaluationService();
 
         } else {
-            service.executeOppEvaluationService();
+            System.enqueueJob(service);
         }
     }
 
+    /***
+    * @description Determines whether Opportunities should be evaluated synchronously or asynchronously
+    * @return Boolean
+    */
+    @TestVisible
+    private Boolean evaluateOppsSynchronously() {
+        RD2_Settings settings = new RD2_Settings();
+
+        if (settings.firstInstallmentAsync == RD2_Constants.FirstInstallmentAsyncOptions.Synchronous_Always.name()) {
+            return true;
+        }
+
+        if (settings.firstInstallmentAsync == RD2_Constants.FirstInstallmentAsyncOptions.Asynchronous_Always.name()
+            && (Limits.getQueueableJobs() < Limits.getLimitQueueableJobs())
+        ) {
+            return false;
+        }
+
+        if (settings.firstInstallmentAsync == RD2_Constants.FirstInstallmentAsyncOptions.Asynchronous_When_Bulk.name()
+            && System.isBatch()
```

Comment:
Rather than just `isBatch` I was thinking when `trigger.new.size()` > 1. This way it catches the scenario where an insert of multiple RD's happens even from outside of a batch job.